### ISSUE TITLE
Flash notice doesn't clear (Turbolinks?)

### ISSUE THREAD

**Issue by hanneskaeufler** [luckyframework/lucky#277]
My understanding is, that flash messages should disappear on page change (I might be wrong?). So when I have an action like this:
```crystal
  post "/posts/create" do
    PostForm.create(params) do |form, post|
      if post
        flash.notice = "Post successfully published."
        redirect Blog::Index
      else
        render Blog::Posts::NewPage, post_form: form
      end
    end
  end
```

I would expect to see this message only for one page.
However I am seeing it for two page reloads. Page reloads being relative of course due to turbolinks.
Am I misunderstanding or is this something we need to improve? Sorry for the lengthy gif.

![dez -05-2017 22-08-55](https://user-images.githubusercontent.com/68024/33630979-66e97f90-da09-11e7-83a7-4c722ccd6455.gif)

**Comment by paulcsmith** [comment]
Thanks for reporting, and the lengthy gif is great! Makes it easy to see exactly what you mean.

I don't think this is Turbolinks. Can you point me to the code that calls `flash`? It may be that you need `flash.now[:notice] =`. If you render a page you need `flash.now`, but if you are redirecting you need `flash`

**Comment by hanneskaeufler** [comment]
Hi, code is already in issue above, so I'm calling `flash.notice = "something"`. First I had `flash[:success] = "something"` but it showed the same behaviour.

**Comment by paulcsmith** [comment]
I should have looked more closely! It's definitely possible that I need to change this behavior for Turbolinks. Thanks for reporting it! For now, I think you can use `flash.now[:notice]` and it should work. Once I fix this I believe you can do what you have.

I think the issue is that a Turbolinks redirect doesn't *actually* redirect:https://github.com/luckyframework/lucky/blob/9e390e12c9f517517f6526d26fde372dfd02585c/src/lucky/redirectable.cr#L12 but I forgot to also change the flash implementation.

**Comment by paulcsmith** [comment]
Ok so this is extra strange. I tried your branch locally and the flash cleared away just fine:

![flash working for me](https://user-images.githubusercontent.com/22394/33637293-1b9b895e-d9ee-11e7-8947-9dbe3a1f0174.gif)

I'm not sure why it would work fine for me, but not for you. Are you using `lucky dev` or are you using `lucky watch`? What OS and browser are you on? Looks like MacOS on chrome.

**Comment by hanneskaeufler** [comment]
Thats ... strange. I am `lucky dev`ing and on the latest macOS and Chrome. Will probably play around with this later to male sure I‘m not seeing ghosts. Thanks for looking into it!

**Comment by hanneskaeufler** [comment]
Luckily my gif showed the bug. But can't reproduce now neither in Chrome nor Safari. So closing this until it shows up again. computers 🤷‍♂️

**Comment by mxie** [comment]
I'm seeing this issue in our project too (regardless of browser) if you want to try it out. Flashes are definitely persisting after navigating to other pages.

**Comment by paulcsmith** [comment]
@mxie yeah I noticed this as well. I found the bug in the flash handler. I’ll try to address it in the next few days :D
